### DISCUSSION TITLE
Petsc compilation

### DISCUSSION THREAD

**Discussion by hbadi** [msys2/MINGW-packages#11765] [General]
Hello,

1 - I try to compile petsc from scratch on msys2 but at the gcc linking stage i get the error "argument list too long".
How do you solve that problem to get the package ?

2 - Is it possible to compile the petsc mingw package with mumps/superlu/umfpack/suitesparse/metis/scotch ?

Best regards.

**Comment by Biswa96** [comment]
petsc is built for mingw packages with this recipe https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-petsc/PKGBUILD

**Reply by hbadi** [reply]
I'm quite new with msys2, how to build a package ?

**Reply by Biswa96** [reply]
I thought you're building the program from git repository. For general package building guidelines, this may help https://www.msys2.org/dev/new-package/#building-the-package
